src/compiler/checker.ts

```diff
@@ -16790,6 +16932,9 @@ namespace ts {
 
                 deferredNodes = [];
                 forEach(node.statements, checkSourceElement);
+                if (isExternalModule(node)) {
+                    checkUnusedImports(node);

```

Comment:
` checkUnusedModuleLocals(node);` here as well. if this is a module, all declarations are local unless exported so for instance:

```ts
export {};  // makes the file a module

var x; // should be reported as unused

export var y; // should not
```